testsuite/cluster-test/src/experiments/mod.rs

```diff
@@ -36,18 +40,63 @@ pub struct Context {
     tx_emitter: TxEmitter,
     prometheus: Prometheus,
     thread_pool_executor: ThreadPoolExecutor,
+    cluster: Cluster,
 }
 
 impl Context {
     pub fn new(
         tx_emitter: TxEmitter,
         prometheus: Prometheus,
         thread_pool_executor: ThreadPoolExecutor,
+        cluster: Cluster,
     ) -> Self {
         Context {
             tx_emitter,
             prometheus,
             thread_pool_executor,
+            cluster,
+        }
+    }
+}
+
+/// Given an experiment name and its flags, it constructs an instance of that experiment
+/// and returns it as a `Box<dyn Experiment>`
+pub fn get_experiment(name: &str, args: &[String], cluster: &Cluster) -> Box<dyn Experiment> {
+    match name {
+        "multi_region_simulation" => Box::new(MultiRegionSimulation::new(
```

Comment:
We can think(in separate PR indeed) how this can be optimized further, so that creation of new experiment does not require change in this file
One option can be introducing macro such as `define_exepriment!("name", ExperimentClass, ExperimentParamClass)` that can be used in file where experiment is defined.
This macro can register this constructor for experiment in some global map, and get_experiment will just simply call it

Other then that LGTM, thanks!

@bors-libra delegate+